checker: use `any` for type casting

<!-- Please make sure to run `v up` before reporting any issues as it may have already been fixed.
     It's also advisable to update all relevant modules using `v outdated` and `v install` -->

<!-- You can use `v doctor` to fill up the next fields -->
```
OS: macos, macOS, 12.4, 21F79
Processor: 8 cpus, 64bit, little endian, Apple M1
CC version: Apple clang version 13.1.6 (clang-1316.0.21.2.5)

getwd: /Users/hunam/Documents/dev/testing-v
vmodules: /Users/hunam/.vmodules
vroot: /Applications/v
vexe: /Applications/v/v
vexe mtime: 2022-07-09 15:02:55
is vroot writable: true
is vmodules writable: true
V full version: V 0.3.0 946da55.1ae11b4

Git version: git version 2.32.1 (Apple Git-133)
Git vroot status: 0.3-82-g1ae11b41
.git/config present: true
thirdparty/tcc status: thirdparty-macos-arm64 173c526e

```

<!-- Please include the standalone minimal reproducible code.
If not, it may be closed by the administrators. -->
**What did you do?**
```v
a := any(123)
```

**What did you expect to see?**
I don't quite know at which point any is a valid type in V.
 
**What did you see instead?**
```c
/tmp/v_501/c.18404743079384999127.tmp.c:2116:4: warning: expression result unused [-Wunused-value]
  (*(int*)_t1.data);
   ^~~~~~~~~~~~~~~
/tmp/v_501/c.18404743079384999127.tmp.c:12017:2: error: use of undeclared identifier 'any'
        any a = ((any)(123));
        ^
1 warning and 1 error generated.
```

--- a/vlib/v/checker/struct.v
+++ b/vlib/v/checker/struct.v
@@ -35,10 +35,6 @@ pub fn (mut c Checker) struct_decl(mut node ast.StructDecl) {
 			}
 		}
 		for i, field in node.fields {
-			if field.typ == ast.any_type {
-				c.error('struct field cannot be the `any` type, use generics instead',
-					field.type_pos)
-			}
 			c.ensure_type_exists(field.typ, field.type_pos) or { return }
 			if field.typ.has_flag(.generic) {
 				has_generic_types = true

--- a/vlib/v/checker/tests/struct_field_with_any_type_err.out
+++ b/vlib/v/checker/tests/struct_field_with_any_type_err.out
@@ -1,4 +1,4 @@
-vlib/v/checker/tests/struct_field_with_any_type_err.vv:2:6: error: cannot use `any` type here, it can only be used in js backend
+vlib/v/checker/tests/struct_field_with_any_type_err.vv:2:6: error: cannot use `any` type here, `any` will be implemented in V 0.4
     1 | struct My_type {
     2 |     fld any
       |         ~~~

--- a/vlib/v/checker/tests/struct_field_with_any_type_err.out
+++ b/vlib/v/checker/tests/struct_field_with_any_type_err.out
@@ -1,4 +1,4 @@
-vlib/v/checker/tests/struct_field_with_any_type_err.vv:2:6: error: struct field cannot be the `any` type, use generics instead
+vlib/v/checker/tests/struct_field_with_any_type_err.vv:2:6: error: cannot use `any` type here, it can only be used in js backend
     1 | struct My_type {
     2 |     fld any
       |         ~~~

--- a/vlib/v/parser/parse_type.v
+++ b/vlib/v/parser/parse_type.v
@@ -607,6 +607,12 @@ pub fn (mut p Parser) parse_any_type(language ast.Language, is_ptr bool, check_d
 					'int_literal' {
 						ret = ast.int_literal_type
 					}
+					'any' {
+						if p.file_backend_mode != .js && p.mod != 'builtin' {
+							p.error('cannot use `any` type here, it can only be used in js backend')
+						}
+						ret = ast.any_type
+					}
 					else {
 						p.next()
 						if name.len == 1 && name[0].is_capital() {

--- a/vlib/v/parser/parse_type.v
+++ b/vlib/v/parser/parse_type.v
@@ -609,7 +609,7 @@ pub fn (mut p Parser) parse_any_type(language ast.Language, is_ptr bool, check_d
 					}
 					'any' {
 						if p.file_backend_mode != .js && p.mod != 'builtin' {
-							p.error('cannot use `any` type here, it can only be used in js backend')
+							p.error('cannot use `any` type here, `any` will be implemented in V 0.4')
 						}
 						ret = ast.any_type
 					}

--- a/vlib/v/parser/tests/cast_to_any_type_err.out
+++ b/vlib/v/parser/tests/cast_to_any_type_err.out
@@ -1,0 +1,6 @@
+vlib/v/parser/tests/cast_to_any_type_err.vv:2:7: error: cannot use `any` type here, it can only be used in js backend
+    1 | fn main() {
+    2 |     a := any(22)
+      |          ~~~
+    3 |     println(a)
+    4 | }

--- a/vlib/v/parser/tests/cast_to_any_type_err.out
+++ b/vlib/v/parser/tests/cast_to_any_type_err.out
@@ -1,4 +1,4 @@
-vlib/v/parser/tests/cast_to_any_type_err.vv:2:7: error: cannot use `any` type here, it can only be used in js backend
+vlib/v/parser/tests/cast_to_any_type_err.vv:2:7: error: cannot use `any` type here, `any` will be implemented in V 0.4
     1 | fn main() {
     2 |     a := any(22)
       |          ~~~

--- a/vlib/v/parser/tests/cast_to_any_type_err.vv
+++ b/vlib/v/parser/tests/cast_to_any_type_err.vv
@@ -1,0 +1,4 @@
+fn main() {
+	a := any(22)
+	println(a)
+}

